source/agora/test/ValidatorGeneral.d

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+
+    Contains tests for general situations where validators participate in
+    nominating a block and reaching consensus.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.ValidatorGeneral;
+
+version (unittest):
+
+import agora.test.Base;
+
+/// Situation: A validator is stopped and wiped clean after the network starts.
+///     And then, two blocks are inserted into the ledger. After the validator
+///     restarts, another new block is in the middle of a consensus.
+/// Expectation: The new block is inserted into the ledger because the validator
+///     has started to validate immediately.
+unittest
+{
+    TestConf conf = { validators : 3, full_nodes : 1 };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    // The node_1 is the validator
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Create two blocks from the Genesis block
+    auto txs = genesisSpendable().enumerate()
+        .map!(en => en.value.refund(WK.Keys[en.index].address).sign())
+        .array();
+    txs.each!(tx => node_1.putTransaction(tx));
+    network.expectBlock(Height(1), 8.seconds);
+
+    txs = txs.enumerate()
+        .map!(en => TxBuilder(en.value)
+            .refund(WK.Keys[8 + en.index].address)
+            .sign())
+        .array();
+    txs.each!(tx => node_1.putTransaction(tx));
+    network.expectBlock(Height(2), 8.seconds);
+
+    // The node_1 restarts, which means that all the date is wiped clean
+    network.restart(node_1);
+    assert(node_1.getBlockHeight() == 0);
```

Comment:
There is a race condition here. The block height is zero but only for a brief moment while the node is loading.

if you add:

```
retryFor(node_1.getBlockHeight() == 2, 5.seconds, node_1.getBlockHeight().to!string);
```

Then you'll see this passes the test.

Therefore I suggest removing the line `assert(node_1.getBlockHeight() == 0);`.

I know that the intention was to test that the node started with an empty blockchain - but we can't have this assert here because it might fail.